planner/cascades/transformation_rules.go

```diff
@@ -14,15 +14,17 @@
 package cascades
 
 import (
+	"github.com/pingcap/tidb/expression"
 	plannercore "github.com/pingcap/tidb/planner/core"
 	"github.com/pingcap/tidb/planner/memo"
+	"github.com/pingcap/tidb/util/ranger"
 )
 
 // Transformation defines the interface for the transformation rules.
 type Transformation interface {
 	GetPattern() *memo.Pattern
-	Match(expr *memo.ExprIter) (matched bool)
-	OnTransform(old *memo.ExprIter) (new *memo.GroupExpr, eraseOld bool, err error)
+	Match(expr *memo.ExprIter) bool
+	OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error)
```

Comment:
Could you add a comment about the return values? For example:
1. in which condition will this function returns more than one `GroupExpr`
2. what does `eraseOld` and `eraseAll` mean